src/bp/ui-studio/src/web/views/OneFlow/diagram/nodes/PromptNode/index.tsx

```diff
@@ -1,49 +1,122 @@
 import {
   StandardIncomingPortModel,
   StandardOutgoingPortModel,
   StandardPortWidget
 } from '~/views/FlowBuilder/diagram/nodes/Ports'
-import { showHeaderV2 } from '~/views/FlowBuilder/diagram/nodes_v2/utils'
 
-import style from '../../../../FlowBuilder/diagram/nodes_v2/style.scss'
+import style from '../Components/style.scss'
+import NodeHeader from '../Components/NodeHeader'
+import NodeWrapper from '../Components/NodeWrapper'
+
+interface Props {
+  node: PromptNodeModel
+  getCurrentFlow: any
+  updateFlowNode: any
+  onDeleteSelectedElements: () => void
+  editNodeItem: (node: PromptNodeModel, index: number) => void
+  selectedNodeItem: () => { node: PromptNodeModel; index: number }
+  getCurrentLang: () => string
+  switchFlowNode: (id: string) => void
+}
 
-import promptStyle from './style.scss'
+const PromptWidget: FC<Props> = ({
+  node,
+  getCurrentFlow,
+  editNodeItem,
+  onDeleteSelectedElements,
+  selectedNodeItem,
+  updateFlowNode,
+  getCurrentLang,
+  switchFlowNode
+}) => {
+  const [error, setError] = useState(null)
+  const [isEditing, setIsEditing] = useState(false)
+
+  const handleContextMenu = e => {
+    e.stopPropagation()
+    e.preventDefault()
+    switchFlowNode(node.id)
+    contextMenu(
+      e,
+      <Menu>
+        <MenuItem
+          text={lang.tr('studio.flow.node.renameBlock')}
+          onClick={() => {
+            setIsEditing(true)
+          }}
+        />
+        <MenuItem
+          text={
+            <div className={style.contextMenuLabel}>
+              {lang.tr('delete')}
+              <ShortcutLabel light keys={['backspace']} />
+            </div>
+          }
+          intent={Intent.DANGER}
+          onClick={onDeleteSelectedElements}
+        />
+      </Menu>
+    )
+  }
 
-class PromptNodeWidget extends React.Component<{ node: PromptNodeModel }> {
-  render() {
-    const node = this.props.node
-    const { type, output } = node.prompt || {}
+  const saveName = (value): void => {
+    setError(null)
 
-    return (
-      <div
-        className={classnames(style.baseNode, promptStyle.nodePrompt, {
-          [style.highlightedNode]: node.isHighlighted
+    if (value) {
+      const alreadyExists = getCurrentFlow().nodes.find(x => x.name === value && x.id !== node.id)
+
+      if (alreadyExists) {
+        setError(lang.tr('studio.flow.node.nameAlreadyExists'))
+        return
+      }
+
+      updateFlowNode({ name: value })
+    }
+
+    setIsEditing(false)
+  }
+
+  const currentLang = getCurrentLang()
+  const selectedContent = selectedNodeItem()
+  const { next, name } = node || {}
+  const { type, output } = node.prompt || {}
+
+  return (
+    <NodeWrapper>
+      <NodeHeader
+        type={type}
+        className={style.prompt}
+        handleContextMenu={handleContextMenu}
+        isEditing={isEditing}
+        saveName={saveName}
+        defaultLabel={lang.tr('studio.flow.node.chatbotPromptsUser')}
```

Comment:
I think the title should also display the type of prompt, otherwise it's pretty much useless (Eg: "Prompt date", "Prompt Number". Short & sweet

Reply:
As discussed on slack, we can revisit with Caro later on for this + the collapsible thing